6.5.group_by.py

```diff
@@ -0,0 +1,5 @@
+def group_by(func, it):
+    return {func(i): [j for j in it if func(i) == func(j)] for i in it}

```

Comment:
Rename 'it' 'i' 'j' Variable names - https://workat.tech/machine-coding/tutorial/writing-meaningful-variable-names-clean-code-za4m83tiesy0 - variable names, abbreviation reference is missing